monix-catnap/jvm/src/test/scala/monix/catnap/SemaphoreJVMSuite.scala

```diff
@@ -45,41 +46,11 @@ abstract class BaseSemaphoreJVMTests(parallelism: Int) extends TestSuite[Schedul
     assert(env.awaitTermination(30.seconds), "env.awaitTermination")
   }
 
-  implicit def contextShift(implicit ec: Scheduler): ContextShift[IO] =
-    SchedulerEffect.contextShift[IO](ec)(IO.ioEffect)
-  implicit def timer(implicit ec: Scheduler): Timer[IO] =
-    SchedulerEffect.timerLiftIO[IO](ec)(IO.ioEffect)
-
   // ----------------------------------------------------------------------------
   val iterations = if (isCI) 1000 else 10000
   val timeout = if (isCI) 30.seconds else 10.seconds
 
-  test("Semaphore (concurrent) — issue #380: — producer keeps its thread, consumer stays forked") { implicit ec =>
```

Comment:
This is an example of what we can't test for with IO anymore. IO is now like if you always did `executeAsync` on Task, will always shift to the pool that backs provided runtime. Not sure if it's required by law tho as I haven't looked into new laws.